### PR TITLE
impl de trait for deserial ansi value and rgb

### PR DESCRIPTION
**Pull request by zwpaper**
deserial json like:

```json
{
    "color_example": "white",
    "ansi_value_example": 255,
     "rgb_example": [255, 255, 255]
}
```

### CODE REVIEW

**Comment by zwpaper** [comment]
Hi @TimonPost , crossterm color use the default deserial implementation, it can not deserial AnsiValue and Rgb, so that I try to implement the deserialization trait to make it support this format:

```json
{
    "color_example": "white",
    "ansi_value_example": 255,
     "rgb_example": [255, 255, 255]
}
```

could you please review this when free

**Review by TimonPost — COMMENTED** [review]
Thanks for the contribution. It would make sense to merge this into crossterm. Just to be sure I understand it right, this works for all kinds of serde crates? 

There is some duplicated test.

File: src/style/types/color.rs [hunk 1]
```diff
@@ -256,3 +336,120 @@ mod tests {
...
+        );
+        assert_eq!(
+            serde_json::from_str::<Color>("\"black\"").unwrap(),
+            Color::Black
+        );
+        assert_eq!(
+            serde_json::from_str::<Color>("\"dark_grey\"").unwrap(),
+            Color::DarkGrey
+        );
+        assert_eq!(
```
[the remark this answers is not in the document]
Comment (zwpaper) [reply src/style/types/color.rs:352]: Good catch! I have also rearranged the cases so that it is easier to catch mistake like that.

**Comment by zwpaper** [comment]
Hi @TimonPost, yes, it should work for all kinds of serde crates, or it would be a bug.

besides the `serde_json` in tests, I would use `serde_yaml` in another project, It could be a validation for that.

**Comment by TimonPost** [comment]
Thanks for the contribution! Its good like this

**Pull request merged**